### PR TITLE
Update devcontainer with java and non-root user

### LINKED ISSUE #644: Failing devcontainer setup in parking sensor sample due to invalid jdk version
## Description

We introduced [a new jdk version](https://github.com/Azure-Samples/modern-data-warehouse-dataops/pull/525/files), but debian bookworm does not have a jdk 11 version.
```sh
1.181 E: Unable to locate package openjdk-11-jdk
````

## Expected Behavior

- Building the devcontainer locally should succeed.

## Reproduce

1. go to the parking sensor e2e sample
2. open vs code
3. run dev container build

fixed by #645

### PR DESCRIPTION
**Pull request by bsherwin**
# Type of PR
- Code changes

## Purpose
Fix issue #644 - Devcontainer failing with unsupported JDK
Fix issue #646 - Devcontainer changes file ownership in linux/wsl
Updates devcontainer to bring Critical and High CVE vulnerabilities reported to zero

## Does this introduce a breaking change? If yes, details on what can break
No

## Validation steps
- Clone branch
- Open VSCode folder in DevContainer
- In devcontainer terminal prompt, verify java version: `java -version`
- In devcontainer terminal prompt, run the command `touch test.file`
- In local WSL prompt, navigate to `/mnt/.../e2e_samples/parking_sensors` (replace "..." with location where you cloned the code)
- Validate you can edit the test.file without requiring root permission.

## Issues Closed or Referenced
- Closes #644, #646

### CODE REVIEW

**Comment by bsherwin** [comment]
@StromFLIX and @cpeeyush: Given you submitted PRs over a year ago, I wanted to include you in the review request. We did a few more updates given CVE vulnerabilities, but your changes/suggestions should be represented in the PR updates. We welcome any feedback from you.

File: e2e_samples/parking_sensors_synapse/.devcontainer/Dockerfile
Comment (ydaponte) [inline e2e_samples/parking_sensors_synapse/.devcontainer/Dockerfile:1]: Let's not include this changes in this PR as this is related with the Synapse sample and it is being retired at the moment.

File: e2e_samples/parking_sensors_synapse/.devcontainer/devcontainer.json
Comment (ydaponte) [inline e2e_samples/parking_sensors_synapse/.devcontainer/devcontainer.json:1]: Same as previous. Let's not mix the Databricks work with the Synapse as it is already being retired.

File: e2e_samples/parking_sensors_synapse/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -3,15 +3,20 @@
...
+				"yzhang.markdown-all-in-one",
+				"ms-azuretools.vscode-bicep"
+			]
+		},
+		"settings": {
+			"python.pythonPath": "/usr/local/bin/python"
+		}
 	},
+	"remoteUser": "mdwuser",
 	"postCreateCommand": "./.devcontainer/post_create.sh"
```
Comment (ydaponte) [inline e2e_samples/parking_sensors_synapse/.devcontainer/devcontainer.json:21]: While testing and I get to the post_create.sh command I get the following error: 
postCreateCommandfailed with exit code 127. Skipping any further user-provided commands. 
Did you also have this problem? Maybe is just a problem on my side, but I would like to confirm with you.

**Review by ydaponte — CHANGES_REQUESTED** [review]
The container started successfully. I left some comments to address and let's have a quick call on the postcreate.sh execution when you read this! Thanks for fixing it!

**Review by ydaponte — APPROVED** [review]
LGTM! Thanks for addressing the comments!

**Pull request merged**